sdk/runanywhere-kotlin/src/jvmMain/kotlin/com/runanywhere/sdk/foundation/device/DeviceInfoService.kt

```
@@ -0,0 +1,73 @@
+    }
+
+    actual fun getTotalMemoryBytes(): Long? {
+        return try {
+            val osBean = ManagementFactory.getOperatingSystemMXBean()
+            if (osBean is com.sun.management.OperatingSystemMXBean) {
+                osBean.totalMemorySize
+            } else {
+                // Fallback to Runtime max memory
+                Runtime.getRuntime().maxMemory()
+            }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)